src/main/java/uk/gov/hmcts/reform/sscs/service/PostponementRequestService.java

```diff
@@ -53,4 +58,27 @@ private void ensureSscsDocumentsIsNotNull(SscsCaseData sscsCaseData) {
         final List<SscsDocument> sscsDocuments = (sscsCaseData.getSscsDocument() == null) ? new ArrayList<>() : sscsCaseData.getSscsDocument();
         sscsCaseData.setSscsDocument(sscsDocuments);
     }
+
+    public void addCurrentHearingToExcludeDates(PreSubmitCallbackResponse<SscsCaseData> response) {
+        SscsCaseData caseData = response.getData();
+        Hearing hearing = caseData.getLatestHearing();
```

Comment:
`getLatestHearings()` has been recently discovered to be a method with side effects. Plus it does additional sorting that may affect performance.

Furthermore, `getLatestHearing()` returns the latest hearing based on its ID, which may not be in chronological order.

- Recommend to check how the removed `setHearingDateToExcludedDate` method was selecting the latest hearing, as it appears to be more stable

Reply:
`getLatestHearings` actually counter-intuitively sorts first by `hearingDateTime` and only afterwards by `id` so the order will be chronological.
As for the side-effects, they are to be handled in the corresponding library. Given that the order is correct, there is therefore no reason to change this code.

Resolving the conversation